arch/x86/pagetables.c

```diff
@@ -121,11 +121,8 @@ static void dump_pagetable(mfn_t table, int level) {
 }
 
 void dump_pagetables(cr3_t *cr3_ptr) {
-#if defined(__x86_64__)
-    int level = 4;
-#else
-    int level = 3;
-#endif
+    int level = PT_LEVELS;
```

Comment:
No need for the indirection, can just pass `PT_LEVELS` directly to the call of `dump_pagetable()`:
```c
    spin_lock(&vmap_lock);
    dump_pagetable(cr3_ptr->mfn, PT_LEVELS);
    spin_unlock(&vmap_lock);
```

Reply:
Fixed, thanks!